lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -306,11 +306,6 @@ def configure(conf)
       # processed by this logging agent.
       @common_labels = determine_agent_level_common_labels
 
-      # For each resource type, there is a list of labels that we want to report
-      # as monitored resource instead of metadata labels. Move them if present.
-      @resource.labels.merge!(
```

Comment:
We are doing the same thing at `agent` level and at `entry` level. The extraction at `agent` level should be redundant because `@label_map` is processed at `entry` level, and we are required to extract these labels after that step anyway. Simply extracting these labels at `entry` level should be sufficient.